packages/gatsby/index.d.ts

```diff
@@ -20,4 +20,6 @@ export interface StaticQueryProps {
 
 export class StaticQuery extends React.Component<StaticQueryProps> {}
 
+export const useStaticQuery: (query: any) => void
```

Comment:
Could probably improve typings here since `query` needs to be graphql tagged template

Reply:
It also should return `Object` (definitely not `void`, as this would probably make TS complain when using result)?

Reply:
Ah, missed that! It can be `null` or obj though

Reply:
@sidharthachatterjee good call! I think I'd just do

```ts
export const useStaticQuery: (query: any) => null | object
```

Reply:
Fixed the return here

I'll fix the input for both `StaticQuery` and `useStaticQuery` in a separate PR

It looks like they should be `TemplateStringsArray` but don't want to touch StaticQuery in this PR